src/TwitterHandler.js

```diff
@@ -0,0 +1,93 @@
+ * functions called without being defined will display a STUB message.
+ * @class
+ * @param  {matrix-appservice-bridge.Bridge}   bridge
+ * @param  {string} roomType  The value as specified in a rooms remote data
+ * (twitter_type) to identify the nature of the room.
+ * @param  {string} aliasPrefix The prefix *after* '#twitter_' to identify alias
+ * requests.
+ */
+var TwitterHandler = function (bridge, roomType, aliasPrefix) {
+  this._bridge = bridge;
+  this.aliasPrefix = "";
+  this.roomType = "";
+  if(typeof aliasPrefix == "string"){
+    this.aliasPrefix = aliasPrefix;
+  }
+  if(typeof roomType == "string"){
+    this.roomType = roomType;
+  }
+}
+
+/**
+ * TwitterHandler.prototype.processInvite - Handler for invites belonging to
+ * a room of this handlers type. The handler should accept or reject the invite
+ * before completing.
+ *
+ * @param  {object} event   The event data of the request.
+ * @param  {object} request The request itself.
+ * @param  {object} context Context given by the appservice.
+ */
+TwitterHandler.prototype.processInvite = function (event, request, context){
+  log.warn("Handler","STUB processInvite");
+  return;
+};
+
+/**
+ * TwitterHandler.prototype.processMessage - Handler for events of type
+ * 'm.room.message'. The handler does not have to act on these.
+ *
+ * @param  {object} event   The event data of the request.
+ * @param  {object} request The request itself.
+ * @param  {object} context Context given by the appservice.
+ */
+TwitterHandler.prototype.processMessage = function (event, request, context) {
+  log.warn("Handler","STUB processMessage");
+  return;
+}
+
+/**
+ * TwitterHandler.prototype.processEvent - Handler for events not of type
+ * 'm.room.message'. The handler does not have to act on these.
+ *
+ * @param  {object} event   The event data of the request.
+ * @param  {object} request The request itself.
+ * @param  {object} context Context given by the appservice.
+ */
+TwitterHandler.prototype.processEvent = function (event, request, context) {
+  log.warn("Handler","STUB processEvent");
+  return;
+}
+
+/**
+ * TwitterHandler.prototype.processAliasQuery - A request to this handler to
+ * provision a room for the given name *after* the global alias prefix.
+ *
+ * @param  {type} name The requested name *after* '#twitter_'
+ * @return {ProvisionedRoom | Promise<ProvisionedRoom,Error>, null}
+ */
+TwitterHandler.prototype.processAliasQuery = function(name){
+  log.warn("Handler","STUB processAliasQuery");
+  return;
+}
+
+
+/**
+ * TwitterHandler.prototype.onRoomCreated - The is called once a room provisoned
+ * by processAliasQuery has been created.
+
+ * @param  {string} alias
+ * @param  {external:RoomBridgeStore.Entry} entry description
+ */
+TwitterHandler.prototype.onRoomCreated = function(alias,entry){
+  log.warn("Handler","STUB onRoomCreated");
+  return;
+}
+
+module.exports = {
+    TwitterHandler: TwitterHandler
+}
```

Comment:
Whilst I appreciate wanting to have a common interface here, I'm not sure how good it is doing. Reading the derived classes, it's not clear which functions were specified in this base class, and all the documentation now resides here as well. You're also assigning properties to each derived handler which then get used thoroughly in each derived class, and it's confusing to try to identify where it was set.

Interfaces are great when you don't know at call-time which actual implementation you want to use, but in your case you really do know at call-time which implementation you want: you call them directly in `TwitterRoomHandler`. This removes any real value in sharing the API between the handlers, so I don't really know what your aim is here.